tests/agent/test_agent.py

```diff
@@ -152,6 +168,16 @@ def test_gx_agent_gets_env_vars_on_init(get_context, gx_agent_config):
     assert agent._config == gx_agent_config
 
 
+def test_gx_agent_gets_default_base_url_if_not_provided(get_context, localhost, monkeypatch):
```

Comment:
Added test for overriding `GX_CLOUD_BASE_URL` with env variable if is available.